packages/kbn-visualization-utils/src/map_vis_to_chart_type.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { ChartType, LensVisualizationType } from './types';
+
+export const mapVisToChartType = (visualizationType: string) => {
+  switch (visualizationType) {
+    case LensVisualizationType.XY:
+      return ChartType.XY;
+    case LensVisualizationType.Metric:
+    case LensVisualizationType.LegacyMetric:
+      return ChartType.Metric;
+    case LensVisualizationType.Pie:
+      return ChartType.Pie;
+    case LensVisualizationType.Heatmap:
+      return ChartType.Heatmap;
+    case LensVisualizationType.Gauge:
+      return ChartType.Gauge;
+    case LensVisualizationType.Datatable:
+      return ChartType.Table;
+  }
+};
```

Comment:
Seeing Marco's comment I actually think he might be right. Changing the code to the one below would make sure that if we add a new lens vis type, typescript will complain. WDYT? 

<img width="644" alt="Screenshot 2024-11-05 at 17 00 33" src="https://github.com/user-attachments/assets/9c5d0d09-6614-4584-9b97-e80818100933">


```suggestion
type ValueOf<T> = T[keyof T];

type LensToChartMap = {
  [K in ValueOf<typeof LensVisualizationType>]: ChartType;
};

const lensTypesToChartTypes: LensToChartMap = {
  [LensVisualizationType.XY]: ChartType.XY,
  [LensVisualizationType.Metric]: ChartType.Metric,
  [LensVisualizationType.LegacyMetric]: ChartType.Metric,
  [LensVisualizationType.Pie]: ChartType.Pie,
  [LensVisualizationType.Heatmap]: ChartType.Heatmap,
  [LensVisualizationType.Gauge]: ChartType.Gauge,
  [LensVisualizationType.Datatable]: ChartType.Table,
};

function isLensVisualizationType(value: string): value is LensVisualizationType {
  return value in LensVisualizationType;
}

export const mapVisToChartType = (visualizationType: string) => {
  if (isLensVisualizationType(visualizationType)) {
    return lensTypesToChartTypes[visualizationType];
  }
};

```

Reply:
Thanx MArta, done here https://github.com/elastic/kibana/pull/197453/commits/6daa2c81d59f5b46da06ab50ab75891ee9bf4860

Reply:
This creates a bug actually